### PR TITLE
Add extra check in fix_job_kwargs

### PR DESCRIPTION
**Pull request by zm711**
For functions that take in an `n_jobs` argument they all run `fix_job_kwargs`, but if the user gives something like 4.0 to mean 4 jobs they end up with (if they have 4 cores)

int(4.0 * 4.0) = 16

So instead we should cast 4.0 -> 4 and just make it 4 jobs no? 

What do you think @alejoe91 ?

### CODE REVIEW

File: src/spikeinterface/core/job_tools.py [hunk 1]
```diff
@@ -87,9 +87,14 @@ def fix_job_kwargs(runtime_job_kwargs):
     n_jobs = job_kwargs["n_jobs"]
     assert isinstance(n_jobs, (float, np.integer, int))
     if isinstance(n_jobs, float):
```
[the remark this answers is not in the document]
Comment (alejoe91) [reply src/spikeinterface/core/job_tools.py:89]: we would need to add another:
```
else:
    n_jobs = int(n_jobs)
```

[the remark this answers is not in the document]
Comment (zm711) [reply src/spikeinterface/core/job_tools.py:89]: So flip the if-else so the more common path is triggered first? 

```python
if isinstance(n_jobs, float) and n_jobs <= 1:
    int(n_jobs * os.cpu_counts())
else:
    int(n_jobs)
```

[the remark this answers is not in the document]
Comment (zm711) [reply src/spikeinterface/core/job_tools.py:89]: I see you're saying fuse all into if-elif-else!

[the remark this answers is not in the document]
Comment (zm711) [reply src/spikeinterface/core/job_tools.py]: Let me know what you think of this!

[the remark this answers is not in the document]
Comment (alejoe91) [reply src/spikeinterface/core/job_tools.py]: Yep looks great!

**Comment by zm711** [comment]
I personally think that we should raise an error if someone sets 0 jobs rather than fix it for them....

**Comment by alejoe91** [comment]
> I personally think that we should raise an error if someone sets 0 jobs rather than fix it for them....
> 
> Should I change that test or revert raising the error?

The old behavior was to set `n_jobs=1` if `n_jobs=0`, right? I agree that raising an error is better ;)
Maybe one wants to type 10 and misses the 1 ;)

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**